autoload/nerdtree/ui_glue.vim

```diff
@@ -509,7 +509,7 @@ endfunction
 
 " FUNCTION: s:openInNewTab(target) {{{1
 function! s:openInNewTab(target)
-    let l:opener = g:NERDTreeOpener.New(a:target.path, {'where': 't'})
+    let l:opener = g:NERDTreeOpener.New(a:target.path, {'reuse': 'all', 'where': 't'})
```

Comment:
@PhilRunninger,

I don't support merging this PR because I don't think the solution here is precise enough.

If I'm editing a file, and it's open in the current tab, pressing `t` on the node for that file in the tree should *force* the file to be opened in a new tab.  Reusing the open window with `o` is okay, but, when I say "Tab Page Please", I think I should get a new tab page!

I think the model for opening the file is okay the way it is, with `o` reusing the window, if it exists, and the other commands forcing specific opening behaviors.

Reply:
To add one more comment:

This is a case where I would prefer a new user preference... something like `g:NERDTreeAlwaysReuseWindows` as opposed to totally changing the behavior users have come to expect. That seems to be the only "precise" solution that avoids changing the way many users expect the NERDTree to behave.

Reply:
Something @scrooloose mentioned a lot is the fact that NERDTree is already bloated, and he was reluctant to add options to satisfy everyone's whims. That's why I didn't make it configurable, but I see your point concerning the change of behavior. I guess I'd be a little put off if I had to change the way I open splits and tabs.

Do you think we should go ahead with the suggested new option, or just close this?

Reply:
I would prefer to just close.  I would only second adding the option if you were to argue that it was genuinely necessary.